docs/releases/4.0.txt

```diff
@@ -191,6 +191,11 @@ CSRF
   this, :ref:`some changes <csrf-trusted-origins-changes-4.0>` to the
   :setting:`CSRF_TRUSTED_ORIGINS` setting are required.
 
+* :class:`~django.middleware.csrf.CsrfViewMiddleware` now uses
+  ``request.META["CSRF_COOKIE_NEEDS_RESET"]`` in place of
+  ``request.META["CSRF_COOKIE_USED"]``, ``request.csrf_cookie_needs_reset``, and
+  ``request.csrf_cookie_set`` to track whether the CSRF cookie should be sent.
```

Comment:
Should there be some wording here to signal that these are internal API's?

Reply:
I added a note.